grant-project/grant-monorepo
# TREZOR & Ledger Address Selection

### What This Does

This is the first step to ledger & trezor auth, allowing you to select an address from both devices. Doesn't do message signing yet though.

### Steps to Test

Requires TREZOR device
* Click Sign In in the top right, select TREZOR
* Accept the popup permission, confirm you're returned to app
* Confirm the addresses you see are correct (Bare in mind that this is testnet addresses)
* Select an address, confirm you can auth as normal
* Go through again, but do something to cause an error (e.g. close popup), confirm you get an error

Requires Ledger device
* Run the app through ssl (e.g. use `https-proxy` or something like that)
* Click Sign In in the top right, select Ledger
* Confirm the addresses you see are correct with Legacy option
* Confirm the addresses you see are correct with Live option
* Select an address, confirm you can auth as normal
* Go through again, but do something to cause an error (e.g. disconnect device), confirm you get an error

## Screenshots

![grant-ledger](https://user-images.githubusercontent.com/649992/46517030-1285c500-c822-11e8-824d-8a610e049e08.gif)
